### PR TITLE
Netty transport websocket

### PR DESCRIPTION
**Pull request by isaacrivriv**

### CODE REVIEW

File: dev/com.ibm.ws.transport.http/bnd.bnd [hunk 1]
```diff
@@ -45,6 +45,8 @@ Export-Package: \
 	com.ibm.ws.http2,\
 	com.ibm.ws.http2.upgrade,\
 	com.ibm.ws.netty.upgrade,\
+	com.ibm.ws.http.netty.inbound,\
```
Comment (isaacrivriv) [inline dev/com.ibm.ws.transport.http/bnd.bnd:48]: Remove this from export

File: dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/dispatcher/internal/channel/HttpRequestImpl.java [hunk 2]
```diff
@@ -125,6 +128,12 @@ public List<HttpCookie> getCookies(String name) {
      */
     @Override
     public List<HttpCookie> getCookies() {
+        List<HttpCookie> cookies = message.getAllCookies();
+        
+        for(HttpCookie c: cookies) {
```
Comment (isaacrivriv) [inline dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/dispatcher/internal/channel/HttpRequestImpl.java:133]: Remove this

File: dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/netty/cookie/CookieDecoder.java [hunk 3]
```diff
@@ -4,14 +4,153 @@
...
+            result = io.netty.handler.codec.http.CookieDecoder.decode(cookieString);
+
+        }
+
+        return result;
+    }
+
 }
+
+//if (foundDollar) {
```
Comment (isaacrivriv) [inline dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/netty/cookie/CookieDecoder.java:105]: Clean up

File: dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/netty/debug/DebugHandler.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
+/*******************************************************************************
```
Comment (isaacrivriv) [inline dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/netty/debug/DebugHandler.java:1]: Remove this

File: dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/netty/inbound/NettyTCPConnectionContext.java [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+    @Override
+    public int getLocalPort() {
+        InetSocketAddress localAddress = (InetSocketAddress) nettyChannel.localAddress();
+        return localAddress.getPort();
+    }
+
+    @Override
+    public SSLConnectionContext getSSLContext() {
+        // TODO Auto-generated method stub
```
Comment (isaacrivriv) [inline dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/netty/inbound/NettyTCPConnectionContext.java:97]: Clean up

File: dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/netty/pipeline/BufferEncoder.java [hunk 6]
```diff
@@ -57,21 +57,19 @@ public void encode(ChannelHandlerContext context, AbstractMap.SimpleEntry<Intege
...
-        if (hasContentLength) {
-            boolean doLastHttpContent = context.channel().attr(NettyHttpConstants.CONTENT_LENGTH).get() == bytesWritten;
-
-            if (!doLastHttpContent) {
-                // Send content with "Content-Length" header
-                System.out.println("Sending http content from encoder!");
-                out.add(new StreamSpecificHttpContent(streamId, Unpooled.wrappedBuffer(message.getWrappedByteBuffer())));
-            } else {
-                // Send last HTTP content
+//        // If "Content-Length" is set, check if it matches the bytes written
```
Comment (isaacrivriv) [inline dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/netty/pipeline/BufferEncoder.java:60]: Clean up

File: dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/netty/pipeline/HttpPipelineInitializer.java [hunk 7]
```diff
@@ -173,7 +182,7 @@ protected void initChannel(Channel channel) throws Exception {
 
         }
 
-//        pipeline.remove(NettyConstants.INACTIVITY_TIMEOUT_HANDLER_NAME);
+        pipeline.remove(NettyConstants.INACTIVITY_TIMEOUT_HANDLER_NAME);
```
Comment (isaacrivriv) [inline dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/netty/pipeline/HttpPipelineInitializer.java:185]: Add read timeout with same behavior as Legacy

File: dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/netty/pipeline/HttpPipelineInitializer.java [hunk 8]
```diff
@@ -213,13 +222,79 @@ private void setupH2Pipeline(ChannelPipeline pipeline) throws NettyException {
      *
      * @param pipeline ChannelPipeline to update as necessary
      */
+//    private void setupH2cPipeline(ChannelPipeline pipeline) {
```
Comment (isaacrivriv) [inline dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/netty/pipeline/HttpPipelineInitializer.java:225]: Clean up

File: dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/netty/pipeline/inbound/HttpDispatcherHandler.java [hunk 9]
```diff
@@ -43,12 +46,22 @@ public HttpDispatcherHandler(HttpChannelConfig config) {
...
     }
 
+    @Override
+    public void handlerAdded(ChannelHandlerContext ctx) {
+        // Store the context for later use
+        context = ctx;
+    }
+
+    // Method to allow direct invocation
+    public void processMessageDirectly(FullHttpRequest request) throws Exception {
```
Comment (isaacrivriv) [inline dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/netty/pipeline/inbound/HttpDispatcherHandler.java:56]: Remove this

File: dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/netty/pipeline/inbound/TransportInboundHandler.java [hunk 10]
```diff
@@ -44,4 +44,10 @@ protected void channelRead0(ChannelHandlerContext context, FullHttpRequest reque
 
     }
 
+    @Override
+    public void channelInactive(ChannelHandlerContext ctx) throws Exception {
```
Comment (isaacrivriv) [inline dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/netty/pipeline/inbound/TransportInboundHandler.java:48]: Remove this

File: dev/com.ibm.ws.webcontainer.servlet.3.1/src/com/ibm/ws/webcontainer31/osgi/srt/SRTConnectionContext31.java [hunk 11]
```diff
@@ -121,27 +121,27 @@ public void finishConnection()
...
-                            upgradedCon.setConnLink(clLink);
-                            upgradedCon.setHttpDisapctherConnLink(dispatcherLink);   
-
-                            upgradedCon.setVirtualConnection(vc);
-                            doUpgradeInit = true;
-                            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
-                                Tr.debug(tc, "finishConnection  webconnection from upgradeHandler "+ upgradedCon.toString());
-                                Tr.debug(tc, "nettyChannel -->"+ tcc +" ,cldevice -->" + cldevice);
-                            }
+                       // if(link.isUsingNetty()) {
```
Comment (isaacrivriv) [inline dev/com.ibm.ws.webcontainer.servlet.3.1/src/com/ibm/ws/webcontainer31/osgi/srt/SRTConnectionContext31.java:124]: Clean up

File: dev/com.ibm.ws.wsoc/bnd.bnd [hunk 12]
```diff
@@ -32,10 +32,13 @@ Export-Package: \
   com.ibm.ws.wsoc.configurator.*;thread-context=true
 
 Import-Package: \
+  io.openliberty.netty.internal*, \
   !*.internal.*, \
   !com.ibm.ws.kernel.boot.cmdline, \
   io.openliberty.wsoc.speclevel, \
   io.openliberty.wsoc.ssl, \
+  com.ibm.ws.http.netty,\
```
Comment (isaacrivriv) [inline dev/com.ibm.ws.wsoc/bnd.bnd:40]: Clean up

File: dev/com.ibm.ws.wsoc/src/com/ibm/ws/wsoc/WsocReadCallback.java [hunk 13]
```diff
@@ -60,6 +60,7 @@ protected void completeLogic(TCPReadRequestContext rrc) {
         ClassLoader originalCL = pushContexts();
 
         try {
+            //popContexts(originalCL);
```
Comment (isaacrivriv) [inline dev/com.ibm.ws.wsoc/src/com/ibm/ws/wsoc/WsocReadCallback.java:63]: Clean up

File: dev/com.ibm.ws.wsoc/src/com/ibm/ws/wsoc/outbound/NettyHttpRequestorWsoc10.java [hunk 14]
```diff
@@ -0,0 +1,524 @@
+/*******************************************************************************
```
Comment (isaacrivriv) [inline dev/com.ibm.ws.wsoc/src/com/ibm/ws/wsoc/outbound/NettyHttpRequestorWsoc10.java:1]: Clean up

File: dev/io.openliberty.netty.internal.impl/src/io/openliberty/netty/internal/tcp/TCPLoggingHandler.java [hunk 15]
```diff
@@ -143,11 +143,23 @@ public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exc
 		super.userEventTriggered(ctx, evt);
 	}
 	
+	@Override
```
Comment (isaacrivriv) [inline dev/io.openliberty.netty.internal.impl/src/io/openliberty/netty/internal/tcp/TCPLoggingHandler.java:146]: Clean up

File: dev/io.openliberty.wsoc.internal_fat/fat/src/io/openliberty/wsoc/common/Constants.java [hunk 16]
```diff
@@ -47,8 +47,8 @@ public enum TimeoutType {
     static int[] directTimeouts = {
                                     15000, //SHORT
                                     15000, //CONNECT
-                                    25000, //DEFAULT
-                                    45000, //LONG
+                                    60000, //DEFAULT
```
Comment (isaacrivriv) [inline dev/io.openliberty.wsoc.internal_fat/fat/src/io/openliberty/wsoc/common/Constants.java:50]: Verify and switch to original timeout

File: dev/io.openliberty.wsoc.internal_fat/fat/src/io/openliberty/wsoc/tests/all/AnnotatedTest.java [hunk 17]
```diff
@@ -52,7 +52,7 @@ public void testMaxSessionIdleTimeout() throws Exception {
         wsocTest.runEchoTest(new AnnotatedClientEP.SessionIdleTest(), "/basic/annotatedIdleTimeout", result, Constants.getLongTimeout());
         long endTime = System.currentTimeMillis();
         long totalTime = endTime - startTime;
-        if ((totalTime < 15000) || (totalTime > 25000)) {
+        if ((totalTime < 15000) || (totalTime > 40000)) {
```
Comment (isaacrivriv) [inline dev/io.openliberty.wsoc.internal_fat/fat/src/io/openliberty/wsoc/tests/all/AnnotatedTest.java:55]: Verify and switch to original timeout

File: dev/io.openliberty.wsoc.internal_fat/fat/src/io/openliberty/wsoc/tests/all/CdiTest.java [hunk 18]
```diff
@@ -418,7 +418,7 @@ public void testCdiInjectWithIdleTimeout() throws Exception {
         wsocTest.runEchoTest(new AnnotatedClientEP.SessionIdleTest(), "/cdi/annotatedIdleTimeoutCDI", result, Constants.getLongTimeout());
         long endTime = System.currentTimeMillis();
         long totalTime = endTime - startTime;
-        if ((totalTime < 15000) || (totalTime > 25000)) {
+        if ((totalTime < 15000) || (totalTime > 40000)) {
```
Comment (isaacrivriv) [inline dev/io.openliberty.wsoc.internal_fat/fat/src/io/openliberty/wsoc/tests/all/CdiTest.java:421]: Verify and switch to original timeout

File: dev/io.openliberty.wsoc.internal_fat/fat/src/io/openliberty/wsoc/tests/all/CdiTest.java [hunk 19]
```diff
@@ -428,7 +428,7 @@ public void testCdiInjectWithIdleTimeout() throws Exception {
         wsocTest.runEchoTest(new AnnotatedClientEP.SessionIdleTest(), "/cdi/annotatedIdleTimeoutCDI", result2, Constants.getLongTimeout());
         endTime = System.currentTimeMillis();
         totalTime = endTime - startTime;
-        if ((totalTime < 3000) || (totalTime > 10000)) {
+        if ((totalTime < 3000) || (totalTime > 20000)) {
```
Comment (isaacrivriv) [inline dev/io.openliberty.wsoc.internal_fat/fat/src/io/openliberty/wsoc/tests/all/CdiTest.java:431]: Verify and switch to original timeout

File: dev/io.openliberty.wsoc.internal_fat/fat/src/io/openliberty/wsoc/tests/all/SessionTest.java [hunk 20]
```diff
@@ -78,7 +78,7 @@ public void testMessageHandlerError() throws Exception {
      */
     public void testSession5() throws Exception {
 
-        int testTimeout = 60000;
+        int testTimeout = 60000*2;
```
Comment (isaacrivriv) [inline dev/io.openliberty.wsoc.internal_fat/fat/src/io/openliberty/wsoc/tests/all/SessionTest.java:81]: Verify and switch to original timeout

File: dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/dispatcher/internal/channel/HttpDispatcherLink.java [hunk 21]
```diff
@@ -220,25 +222,47 @@ public NettyServletUpgradeHandler prepareForUpgrade() {
             System.out.println("Removing keepalive handler on servlet upgrade!");
             nettyContext.channel().pipeline().remove(handler);
         }
+
```
Comment (isaacrivriv) [inline dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/dispatcher/internal/channel/HttpDispatcherLink.java:225]: Clean up

File: dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/channel/internal/HttpServiceContextImpl.java [hunk 22]
```diff
@@ -2251,25 +2264,42 @@ final protected void sendHeaders(HttpBaseMessageImpl msg) throws IOException {
...
+        if (isSwitching && "websocket".equalsIgnoreCase(response.headers().get(HttpHeaderNames.UPGRADE))) {
+            nettyContext.channel().attr(NettyHttpConstants.PROTOCOL).set("WebSocket");
+        }
         this.nettyContext.channel().writeAndFlush(this.nettyResponse);
-
+        
         this.setHeadersSent();
+
+
+        try {
```
Comment (isaacrivriv) [inline dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/channel/internal/HttpServiceContextImpl.java:2294]: Clean up

**Review by mrsaldana — APPROVED** [review]
Reviewed in Slack. Cleanup to commented lines and sysouts to be done in future PRs.

**Pull request merged**